File: Problemas/158A - Next Round/158A_Next_Round.cpp
// 158A - Next Round
// https://codeforces.com/problemset/problem/158/A


#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
using namespace std;

int main(){

    // Input:
    // 1) num_participantes = Número de participantes
    // 2) k_participante = K-esimo participante
    int num_participantes, k_particpante;
    cin >> num_participantes >> k_particpante;

    // Input:
    // 3) lista_participantes = Arreglo de puntuaciones de participantes
    int lista_participantes[100];
    for(int i = 0; i < num_participantes; i++) {
        cin >> lista_participantes[i];
    }

    // Puntuación del k-esimo participante
    int k_puntuacion = lista_participantes[k_particpante - 1];

    // Número de participantes que pasan a la siguiente ronda
    int participantes_siguiente_ronda = 0;

    // Compara las puntuaciones de los particpantes contra la del k-esimo participante
    for(int participante = 0; participante < num_participantes; participante++) {
        if(lista_participantes[participante] >= k_puntuacion && lista_participantes[participante] > 0) {
            participantes_siguiente_ronda++;
        }
    }

    // Imprime el número de participantes que pasan a la siguiente ronda
    cout << participantes_siguiente_ronda;

    return 0;
}

File: Problemas/158A - Next Round/158A_Next_Round.py
# 158A - Next Round
# https://codeforces.com/problemset/problem/158/A

# Inputs:
# 1) num_participantes = Número de participantes
# 2) k_participante = K-esimo participante
num_participantes, k_participante = map(int, input().split())

# Input:
# 3) lista_participantes = Arreglo de puntuaciones de participantes
lista_puntuaciones = input().split()

# Puntuacion del k-esimo participante
k_puntuacion = int(lista_puntuaciones[k_participante - 1])

# Numero de participantes que pasan a la siguiente ronda
ronda_participantes = 0

# Compara todas las puntuaciones contra la del k-esimo participante
for participante in range(0, num_participantes):
    if(int(lista_puntuaciones[participante]) >= k_puntuacion and int(lista_puntuaciones[participante]) > 0):
        ronda_participantes += 1

# Imprime el número de participantes que pasan a la siguiente ronda
print(ronda_participantes)
